### PR TITLE
Use features to choose cryptography algorithms.

### PR DESCRIPTION
**Pull request by chaoticlonghair**
Add features to some Cargo.toml files and update the CI scripts.

### CODE REVIEW

**Comment by zhangsoledad** [comment]
@Infrastructure

**Comment by zhangsoledad** [comment]
@cryptape/infrastructure

**Comment by u2** [comment]
Why not create a new branch, but fork to a new repo

File: engine/Cargo.toml [hunk 1]
```diff
@@ -11,3 +11,15 @@ cita-crypto = { path = "../cita-crypto" }
 engine_json = { path = "../engine_json" }
 proof = { path = "../proof" }
 pubsub = { path = "../pubsub" }
+
+[features]
+default = []
+secp256k1 = ["cita-crypto/secp256k1", "libproto/secp256k1", "proof/secp256k1", "engine_json/secp256k1"]
+ed25519 = ["cita-crypto/ed25519", "libproto/ed25519", "proof/ed25519", "engine_json/ed25519"]
```
Comment (u2) [inline engine/Cargo.toml:18]: 看着好复杂，后面在cita库里面会不是越写越多？

**Review by u2 — APPROVED** [review]

File: cita-crypto/Cargo.toml [hunk 2]
```diff
@@ -10,7 +10,10 @@ cita-sm2 = { path = "../cita-sm2", optional = true  }
 util = { path = "../util" }
 
 [features]
-default = ["secp256k1"]
-ed25519 = ["cita-ed25519"]
-secp256k1 = ["cita-secp256k1"]
-sm2 = ["cita-sm2"]
+default = []
```
Comment (yangby-cryptape) [inline cita-crypto/Cargo.toml]: **TODO** Need an optimization for dependencies.

File: cita-crypto/Cargo.toml [hunk 3]
```diff
@@ -10,7 +10,10 @@ cita-sm2 = { path = "../cita-sm2", optional = true  }
...
+secp256k1 = ["cita-secp256k1", "cita-secp256k1/sha3hash"]
+ed25519 = ["cita-ed25519", "cita-ed25519/blake2bhash"]
+sm2 = ["cita-sm2", "cita-sm2/sm3hash"]
+sha3hash = ["util/sha3hash"]
+blake2bhash = ["util/blake2bhash"]
+sm3hash = ["util/sm3hash"]
```
Comment (yangby-cryptape) [inline cita-crypto/Cargo.toml]: **TODO** Need an optimization for these dependencies.

File: cita-crypto/Cargo.toml [hunk 4]
```diff
@@ -10,7 +10,10 @@ cita-sm2 = { path = "../cita-sm2", optional = true  }
...
 [features]
-default = ["secp256k1"]
-ed25519 = ["cita-ed25519"]
-secp256k1 = ["cita-secp256k1"]
-sm2 = ["cita-sm2"]
+default = []
+secp256k1 = ["cita-secp256k1", "cita-secp256k1/sha3hash"]
+ed25519 = ["cita-ed25519", "cita-ed25519/blake2bhash"]
+sm2 = ["cita-sm2", "cita-sm2/sm3hash"]
+sha3hash = ["util/sha3hash"]
```
Comment (chaoticlonghair) [inline cita-crypto/Cargo.toml:17]: **TODO** Need an optimization for these dependencies.

**Pull request merged**